.nvmrc

```diff
@@ -1 +1 @@
-16.19.0
\ No newline at end of file
+18.20.1
```

Comment:
Considering the repo node version is being updated, it'd be good to also update the API docker image to [18-alpine3.19](https://hub.docker.com/layers/library/node/18-alpine3.19/images/sha256-05583a00b42e064e4f0bb05fb2392133aa2a8728f50d16bd95ccc705ddf96bc9?context=explore) (leaving the dev version mismatched with the deployment could cause confusing/hidden behavior if the node version breaks anything).

Is there anything else that should be updated when changing this?

Reply:
Hmm, is updating the entire repo node version required? 

There is (was?) an issue with newer node versions and sharp, the image processor that we use, see here https://github.com/funmusicplace/mirlo/issues/471

Reply:
Yeah - it looks like the unit test run on this PR might've run into a similar issue.

I've done some poking at the docker images locally - it appears that the `volumes: ./:/var/www/api` mount is the cause, which replaces the `node_modules` that should be populated by `yarn install` on the container (hence the docker image does not have the right version of sharp installed). That said, while removing  this has fixed the sharp error, I've run into a few more issues as a result - so I'll see where I get with debugging that.

It shouldn't be required to update the entire repo's node version - so that might work as a temporary fix if I don't get anywhere. That said, [Node 16 is nearing the end of its maintenance status](https://nodejs.org/en/about/previous-releases) so I think this upgrade will still be needed at some point.

Reply:
> it appears that the volumes: ./:/var/www/api mount is the cause, which replaces the node_modules that should be populated by yarn install on the container (hence the docker image does not have the right version of sharp installed)

It's entirely possible I don't know how docker works but do we not need to mount our code inside the docker container somehow? Maybe I'm misunderstanding. 

Either way, if you get it working I'll be grateful! Docker = not my forte.

Reply:
Yeah! When the docker *image* is built, the `COPY . .` command will place every file from the build location into the image (aside from those in `.dockerignore` - which includes `node_modules`).

When the docker *container* is created, it uses the most recent docker image build, and mounts any volumes that are specified.

Since you're using nodemon, it's seems like these volume mounts are intended to sync changes on the host with the container for "live reloading" without rebuilding the image. With the `node_modules` change, this *could* still be done to a limited extent - but any changes to `package.json` requiring another `yarn install` would now need the image to be rebuilt.

Are these dockerfiles (and nodemon) also being used in production? If so, it might make sense to separate these configs into a "dev" config that uses nodemon, and a "deploy" that does not - which might simplify this config a little.

Reply:
> Are these dockerfiles (and nodemon) also being used in production? If so, it might make sense to separate these configs into a "dev" config that uses nodemon, and a "deploy" that does not - which might simplify this config a little.

They're not used on production (though ideally they would be I think), production runs on render, which uses the render.yaml. 

I think it'd be great to have a docker file that is able to be set up in a production environment.

Reply:
Hey! So I have a follow-up question to this - I've noticed that render.yaml has an `env: static` service that hosts the client build, but the actual website seems to be on Netlify. Does that mean the render web service is unused?

Reply:
It is yeah. We switched to netlify for static stuff. It's currently disabled in render itself. We could remove it, but I feel like there's an argument for leaving it if anyone wants to use Render to spin things up themselves with minimal tweaking (a kind of "install with render" button)